### PR TITLE
feat: add freerenewal

### PR DESCRIPTION
**Pull request by fricoben**
## Summary by CodeRabbit

- **New Features**
  - Introduced `ArCurrencyDropdown` component for currency selection.
  - Added `useNeedAllowances` and `useNeedSubscription` hooks for managing token allowances and subscriptions.
  - Added new enums `ArCurrency` and `ArCurrencyIcon` for currency handling.

- **Enhancements**
  - Updated `RegisterCheckboxes` and `RegisterSummary` to support optional renewal properties and multiple currencies.
  - Added new types and structures for handling subscriptions and token allowances.

- **Bug Fixes**
  - Corrected dependencies in `CheckoutCard` component.
  - Refactored `useBalances` for better balance data handling.

- **Refactor**
  - Renamed `FreeRenewal` to `FreeRenewalDiscount`.
  - Updated `freeRenewal` functions to include new parameters.

- **Chores**
  - Updated `next` package version in `package.json`.
  - Added test cases for `areArraysEqual` and `processSubscriptionData` functions.

### CODE REVIEW

File: hooks/useNeedSubscription.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          setNeedSubscription((prevState) => ({
+            ...prevState,
+            ...newNeedSubscription,
+          }));
+        });
+    }
+  }, [address]);
+
+  return needSubscription;
+}
```
[the remark this answers is not in the document]
Comment (fricoben) [reply hooks/useNeedSubscription.tsx]: What code do you propose to do so

[the remark this answers is not in the document]
Comment (fricoben) [reply hooks/useNeedSubscription.tsx]: make it a typescript code

[the remark this answers is not in the document]
Comment (fricoben) [reply hooks/useNeedSubscription.tsx]: Now create a jest test following the test format of this repo for the functions `fetchNonSubscribedDomains ` & `processSubscriptionData `

[the remark this answers is not in the document]
Comment (fricoben) [reply hooks/useNeedSubscription.tsx]: rewrite `subscriptionUtils.ts` with commentaries giving explaination in the `forEach ` loops

**Comment by fricoben** [comment]
## E2E Test cases

- [x] A domain with 0 allowances both token and AR
- [x] A domain with 1 allowance ETH token and 0 AR (test with all currencies ETH/STRK/DOUBLE)
- [x] A domain with 1 allowance ETH AR and 0 token (test with all currencies ETH/STRK/DOUBLE)
- [x] A domain with 1 allowance STRK token and 0 AR (test with all currencies ETH/STRK/DOUBLE)
- [x] A domain with 1 allowance STRK AR and 0 token (test with all currencies ETH/STRK/DOUBLE)
- [x] Same as the last four tasks but with several domains

File: types/backTypes.d.ts [hunk 2]
```diff
@@ -88,3 +88,18 @@ type RenewalData = {
...
+  enabled: boolean;
+  allowance: string;
+  renewer_address: string;
+  auto_renew_contract: AutoRenewalContracts | null;
+  token: ERC20Contract;
+};
+
+type SubscriptionInfos = {
+  [key: string]: {
+    eth_subscriptions: Subscription[];
```
Comment (irisdv) [inline types/backTypes.d.ts]: `eth_subscriptions` could also be null no ? in the api response it can be null

**Review by irisdv — APPROVED** [review]
lgtm ! Tested all the cases listed and it worked. I just left a small comment regarding typing.

**Pull request merged**